pywinauto/controls/uia_controls.py

```diff
@@ -566,6 +566,37 @@ def __init__(self, elem):
         except NoPatternInterfaceError:
             self.iface_grid_support = False
 
+        self.is_table = not self.iface_grid_support and self.element_info.control_type == "Table"
+        self.row_header = False
+        self.col_header = False
+
+    def __raise_not_implemented(self):
+        raise NotImplementedError("This method not work properly for WinForms DataGrid, use cells()")
+
+    def __update_row_header(self):
+        try:
+            self.row_header = all(isinstance(row.children()[0], HeaderWrapper) for row in self.children())
```

Comment:
This implementation is more optimal:
```python
try:
    self.row_header = all(isinstance(row.iter_children().next(), HeaderWrapper) for row in self.children())
except StopIteration:
    self.row_header = False
```
Similar optimizations can be done for all `children()[0]` calls.